src/features/postsSlice.ts

```diff
@@ -0,0 +1,68 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type PostsState = {
+  posts: Post[];
+  loaded: boolean;
+  hasError: boolean;
+  error: string;
+  selectedPost: Post | null;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+  error: '',
+  selectedPost: null,
+};
+
+export const init = createAsyncThunk('posts/fetch', (userId: number) => {
+  return getUserPosts(userId);
+});
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    setPosts: (state, action) => {
+      return {
+        ...state,
+        posts: action.payload,
+      };
+    },
+    selectedPost: (state, action) => {
```

Comment:
The reducer name 'selectedPost' could be more descriptive, for example 'setSelectedPost', to better indicate its function.